### DISCUSSION TITLE
WiFi configuration for bare metal

### DISCUSSION THREAD

**Discussion by chris102994** [siderolabs/talos#6911] [Q&A] [answered]
I have some Raspberry Pi and Dell Wyse 5070s that have integrated WiFi and was attempting to configure the [Network Configuration](https://www.talos.dev/v1.3/reference/configuration/#networkconfig) to utilize the WiFi rather than solely relying on Ethernet but I am unable to find any documentation or examples of how to implement this other than point the config at the WiFi device (i.e. no fields for SSID or PSK). 

If this is possible could you provide me with an example and if not are there any plans to implement this?

**Comment by smira** [comment, marked as answer]
Talos doesn't support any wireless network configuration, and it lacks components to do WiFi networking.

**Reply by chr0n1x** [reply]
sorry to bump this, but will this be supported at any point?

**Reply by smira** [reply]
There's no plan so far to support WiFi.

**Reply by vasiliy-mikhailov** [reply]
I'm disappointed to learn that Talos OS doesn't support WiFi and has no plans to add it. For entry-level homelab enthusiasts like me, who are often on a tight budget, this is a significant barrier. Many of us can't afford extra hardware like switches, routers, or long Ethernet cables just to test a new system. WiFi support would make Talos much more accessible for experimenting in resource-constrained setups.

I was excited to try Talos earlier in my Kubernetes journey—my queue was Minikube (done), MicroK8s (done), Kubespray (done), Deckhouse (done), Kubernetes the Hard Way (current), Kubeadm, k3s, and Talos. I initially planned to test Talos before Kubespray, but the lack of WiFi support pushed it to the end of the list. I hope the Talos team reconsiders WiFi support to make it easier for beginners to explore this promising OS.

**Reply by rothgar** [reply]
We appreciate your comment and would welcome contributions to make wifi work in an API driven way that fits the Talos ideology.

Sidero Labs is a small team on a tight time budget and we want to make Talos maintainable long term. Because of that goal and restriction we don't have the development resources to create features that will not drive the company towards financial sustainability or make the project easier to maintain.

We love our homelabbers, but this project wouldn't exist without paying customers. If a company is willing to sponsor the work to make wifi configurable in Talos we will absolutely consider it.

For now, we suggest using an ethernet to wireless bridge which can be purchased online for under $100 USD in most online stores and do not require any changes or configuration from Talos.

**Reply by chr0n1x** [reply]
@vasiliy-mikhailov I empathize with your scenario. That being said - are you running VMs? If you have extra compute via Proxmox or even VMWare/equivalent, Talos has been extremely easy to set up for me, and I would encourage you to just jump in.

Personally, while testing things on my own proxmox machine I was able to spin up 17 (_literally_ 17!) new instances over and over in less than a few hours just to test out various permutations of modules and configurations for my cluster. And my proxmox instance does _not_ have wifi.

**Reply by Serisium** [reply]
My company has been evaluating Talos to manage our fleet of remote kiosk-style devices, but the decision to exclude WiFi kernel support and configuration options makes Talos unviable for our case. 

Everything else about Talos on bare-metal machines is perfect for us, and I understand the reasoning behind excluding WiFi. I'll keep an eye on the project and plan to revisit it if/when WiFi is added. In the meantime, keep up the great work!
